downloader.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import { https } from 'follow-redirects';
+import MemoryStream from 'memorystream';
+import { keccak256 } from 'js-sha3';
+
+function getVersionList () {
+  console.log('Retrieving available version list...');
+
+  return new Promise<string>((resolve, reject) => {
```

Comment:
Any reason to directly use promises rather than `async`/`await` here?

Reply:
Yes. As far as I know the `https.get`module from nodejs is not "awaitable" so I needed to wrap it in a promise to explicitly resolve the response.